### ISSUE TITLE
random.uniform((),minval,maxval) returns array instead of scalar tensor when min or maxval is not a scalar tensor

### ISSUE THREAD

**Issue by Pixel-Therapy** [tensorflow/tensorflow#34363]
```
>>> import tensorflow as tf
>>> scalar = tf.zeros(shape=())
>>> array = tf.zeros(shape=(1,))

>>> tf.random.uniform(shape=(),minval = scalar)
<tf.Tensor: id=25, shape=(), dtype=float32, numpy=0.021499991>

>>> tf.random.uniform(shape=(),minval = array)
<tf.Tensor: id=31, shape=(1,), dtype=float32, numpy=array([0.9388697], dtype=float32)>
```
Expected behavior is to either trow an error or treat single element tensor as scalar and return a scalar.

-win10, tf2, cuda

**Comment by rmothukuru** [comment]
@Pixel-Therapy,
Correct me if I am wrong but, From this [TF Link](https://www.tensorflow.org/api_docs/python/tf/random/uniform) and [The Source Code](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/ops/random_ops.py#L186-L252),  it is mentioned that `random.uniform()` returns a Tensor but it is not mentioned that it will be a Scalar Tensor.

Can you please provide the reference where it mentions `random.uniform` returns a Scalar Tensor. Thanks!

**Comment by Pixel-Therapy** [comment]
> @Pixel-Therapy,
> Correct me if I am wrong but, From this [TF Link](https://www.tensorflow.org/api_docs/python/tf/random/uniform) and [The Source Code](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/ops/random_ops.py#L186-L252), it is mentioned that `random.uniform()` returns a Tensor but it is not mentioned that it will be a Scalar Tensor.
> 
> Can you please provide the reference where it mentions `random.uniform` returns a Scalar Tensor. Thanks!

The first argument in random.uniform asks for shape, ill edit my example as shape = ().
It seems that random.uniform always follows this shape except in the above case.

**Comment by yongtang** [comment]
The issue is that in one code path, math_ops.add was called which implicitly broadcast.

Created a PR #34399 for the fix.

**Comment by yongtang** [comment]
This issue is still being worked on. PR #34399 depends on PR #38544. Once PR #38544 is merged, PR #34399 will be reopened and this issue will be fixed by then.

**Comment by yongtang** [comment]
PR #38544 has been merged. However, the issue is not fixed yet. Will re-submit the PR  of PR #34399 to eventually fix this issue.
